File: apps/python-gateway/test_ws.py
import asyncio
import websockets

async def test():
    try:
        async with websockets.connect("ws://127.0.0.1:8000/v1/ws/openclaw") as ws:
            print("Connected!")
            await ws.send('{"type": "ping"}')
            print(f"Received: {await ws.recv()}")
    except Exception as e:
        print(f"Error: {e}")

asyncio.run(test())

File: plugins/ts-openclaw-channel/test-ws.mjs
import WebSocket from 'ws';

const ws = new WebSocket('ws://127.0.0.1:8000/v1/ws/openclaw', {
  headers: { "x-openclaw-id": "openclaw-test" }
});

ws.on('open', () => {
  console.log('connected');
  ws.send(JSON.stringify({ type: 'ping' }));
});

ws.on('message', (data) => {
  console.log('received:', data.toString());
  ws.close();
});

ws.on('error', (err) => {
  console.error('error:', err);
});
